src/bin/ion/commands/generate/model.rs

```diff
@@ -332,6 +369,12 @@ pub struct Scalar {
     source: IslType,
 }
 
+impl Scalar {
+    pub fn base_type(&self) -> &FullyQualifiedTypeReference {
```

Comment:
All the properties of these `AbstractDataType`s are available at the template level using the serialized form of these types. i.e. on the template side you can just use `scalar_model.base_type`. That way you don't really need to include these methods unless you are using it in the generator in which case it actually makes it difficult to have the separation between the generator and templates.